sort/topo_sort.py

```diff
@@ -19,16 +19,21 @@ def do_topo_sort(
     for level in sorted_dependencies:
         temp_mod_dict = {}
         for package_id in level:
-            temp_mod_dict[package_id] = active_mods_json[package_id]
+            for uuid, mod_data in active_mods_json.items():
+                mod_package_id = mod_data["packageId"]
+                if package_id == mod_package_id:
+                    temp_mod_dict[uuid] = active_mods_json[uuid]
```

Comment:
Technically yes - however `get_active_inactive_mods` and `get_active_mods_from_config` should be taking care of this in regards to the data being used in sorting (the only mod_data in items() is data from the active_mods_json

Now that we are factoring those duplicates there with the tracking dict in `self.duplicate_mods` this is actually a non-issue (i.e. you will have the correct mod loaded as long as you don't have more than one instance of that mod per data_source)

However, I do agree that whatever fix above should definitely translate here (although it won't change the result, the logic will be more accurate because of the change)

Reply:
Fixed